website/scripts/get-average-bug-and-pr-age.js

```diff
@@ -0,0 +1,168 @@
+module.exports = {
+
+
+  friendlyName: 'Deliver Fleet GitHub repo statistics',
+
+
+  description: 'Get the average age of open pull requests and issues with the bug label in the fleetdm/fleet GitHub repo.',
+
+
+  exits: {
+
+
+  },
+
+
+  fn: async function () {
+
+    sails.log('Getting average open time for issues with the "bug" label and open pull requests in the fleetdm/fleet Github repo...');
+
+
+    let baseHeaders = {
+      'User-Agent': 'Fleet average open time',
+    };
+
+    const oneDayInMs = (1000 * 60 * 60 * 24);
+    const todaysDate = new Date;
+
+
+    //   ██████╗ ██████╗ ███████╗███╗   ██╗    ██████╗ ██╗   ██╗ ██████╗ ███████╗
+    //  ██╔═══██╗██╔══██╗██╔════╝████╗  ██║    ██╔══██╗██║   ██║██╔════╝ ██╔════╝
+    //  ██║   ██║██████╔╝█████╗  ██╔██╗ ██║    ██████╔╝██║   ██║██║  ███╗███████╗
+    //  ██║   ██║██╔═══╝ ██╔══╝  ██║╚██╗██║    ██╔══██╗██║   ██║██║   ██║╚════██║
+    //  ╚██████╔╝██║     ███████╗██║ ╚████║    ██████╔╝╚██████╔╝╚██████╔╝███████║
+    //   ╚═════╝ ╚═╝     ╚══════╝╚═╝  ╚═══╝    ╚═════╝  ╚═════╝  ╚═════╝ ╚══════╝
+    //
+
+
+    let allIssuesWithBugLabel = [];
+    let pageNumberForPossiblePaginatedResults = 1;
+    let maximumResultsPerRequest = 100;
+    let daysSinceBugsWereOpened = [];
+
+    // Fetch all open issues in the fleetdm/fleet repo with the bug label.
+    let issuesWithBugLabel = await sails.helpers.http.get(
+      `https://api.github.com/repos/fleetdm/fleet/issues`,
+      {
+        'state': 'open',
+        'labels': 'bug',
+        'per_page': maximumResultsPerRequest,
+        'page': pageNumberForPossiblePaginatedResults,
+      },
+      baseHeaders
+    ).retry();
+
+    allIssuesWithBugLabel = allIssuesWithBugLabel.concat(issuesWithBugLabel);
+    // Set a flag that will be true if the number of results we recieved is the same amount of results that are sent at one time, if this is true, we will request the next set of results until this value is false.
+    let maybeNeedToCheckForMoreBugs = (issuesWithBugLabel.length === maximumResultsPerRequest);
+
+    while(maybeNeedToCheckForMoreBugs) {
```

Comment:
Might be able to do this with a method in sails.helpers.flow (I think there's a "while") and that'll automatically include a timeout as a failsafe against endless looping.  Otherwise, you'll want to build in a counter (e.g. no more than 100 pages, else give up)

Reply:
I updated this to use `sails.helpers.flow.until` with a timeout of 10000 milliseconds